### PR TITLE
add option to print product/vendor IDs

### PR DESCRIPTION
**Pull request by durka**
This adds a convenience feature to print the device descriptor in the list, to see what is connected to each port.

I couldn't find any way to do it except searching the array (I tried to use `LIBUSB_REQUEST_GET_DESCRIPTOR` but it just filled my struct with zeros).

### CODE REVIEW

File: uhubctl.c [hunk 1]
```diff
@@ -278,7 +281,31 @@ static int hub_port_status(struct libusb_device * dev, int nports, int portmask)
                 break;
             }
 
-            printf("   Port %d: %04x", port, port_status);
+            if (opt_verbose == 1) {
+                struct libusb_device *devi;
+                struct libusb_device_descriptor descr = { 0 };
```
Comment (mvp) [inline uhubctl.c]: This causes compilation warning. Just avoid initialization completely:

    struct libusb_device_descriptor descr;

**Comment by durka** [comment]
Hmmmmm. It does work on Linux. I will try to see what the deal is on Mac in
the next few days.

Yes I thought of adding another option for lookup by ID. I think it would
be useful. At least for now (if this feature is working) you can list, grep
for your device, then invoke uhubctl again.

**Comment by durka** [comment]
So it seems to be sort of working for me on OSX? Here is what I get (this is a Macbook Air with no peripherals plugged in, so these are internal or simulated hubs):

```
$ ./uhubctl -iV
Current status for hub 29-1.8, vendor 0424:2513, 3 ports
   Port 1 (no device): 0107 power suspend enable connect
   Port 2 (05ac:024c): 0103 power enable connect
   Port 3 (no device): 0507 highspeed power suspend enable connect
Current status for hub 29-1, vendor 8087:0024, 8 ports
   Port 1 (no device): 0100 power
   Port 2 (no device): 0100 power
   Port 3 (no device): 0100 power
   Port 4 (no device): 0100 power
   Port 5 (no device): 0100 power
   Port 6 (no device): 0100 power
   Port 7 (no device): 0100 power
   Port 8 (0424:2513): 0503 highspeed power enable connect
```

Device `05ac:024c` is the keyboard. So numbers are missing for two devices on the first hub. They must be some other built-in devices like the touchpad, camera, or card reader.

Here is the full output of `ioreg -p IOUSB -l -w 0`: [gist](https://gist.github.com/b3ba6156b2e9fb6cd71493e0af3a2814)

**Comment by mvp** [comment]
You seem to be running older Mac OS version. I tried this on latest MacOS Sierra, and it doesn't work. This can be related to issue #33...

**Comment by xayide1788** [comment]
Would it be possible to include vendor and product string like in lsusb?

Bus 005 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 004 Device 004: ID 05ac:0221 Apple, Inc. Aluminum Keyboard (ISO)
Bus 004 Device 003: ID 045e:0047 Microsoft Corp. IntelliMouse Explorer 3.0
Bus 004 Device 002: ID 05ac:1006 Apple, Inc. Hub in Aluminum Keyboard
Bus 004 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 003 Device 009: ID 0471:0329 Philips (or NXP) SPC 900NC PC Camera / ORITE CCD Webcam(PC370R)
Bus 003 Device 008: ID 0a12:0001 Cambridge Silicon Radio, Ltd Bluetooth Dongle (HCI mode)
Bus 003 Device 007: ID 0b9b:0904 Dipl.-Ing. Stefan Kunde 
Bus 003 Device 010: ID 10c4:ea60 Cygnal Integrated Products, Inc. CP210x UART Bridge / myAVR mySmartUSB light
Bus 003 Device 005: ID 0403:6001 Future Technology Devices International, Ltd FT232 USB-Serial (UART) IC
Bus 003 Device 004: ID 1941:8021 Dream Link WH1080 Weather Station / USB Missile Launcher
Bus 003 Device 003: ID 04cc:1521 ST-Ericsson USB 2.0 Hub
Bus 003 Device 002: ID 8087:0024 Intel Corp. Integrated Rate Matching Hub
Bus 003 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub

**Comment by mvp** [comment]
Yes, @xayide1788, thats the basic idea - everyone wants that.
I think just showing vid:pid is too limited, we should show more.
This has few challenges: 
* avoid using too long strings for output
* not break scripts that might be parsing uhubctl output
* make it work on all plaforms (Mac OS Sierra is broken atm)

**Comment by mvp** [comment]
I have implemented to always print needed info in 222f32cf86bdcb438a4b7cde24358ff652d22bc8.
This is now obsolete, closing.

**Pull request closed without merging**